nixos/modules/services/video/broadcast-box.nix

```diff
@@ -0,0 +1,239 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  inherit (lib)
+    mkIf
+    mkEnableOption
+    mkPackageOption
+    mkOption
+    hasAttr
+    attrNames
+    types
+    match
+    optional
+    ;
+  inherit (cfg) settings;
+  cfg = config.services.broadcast-box;
+
+  addressToPort = address: lib.toInt (lib.last (lib.splitString ":" address));
+  httpPort = addressToPort settings.HTTP_ADDRESS;
+  tcpMuxPort = addressToPort settings.TCP_MUX_ADDRESS;
+  httpRedirect = (settings.ENABLE_HTTP_REDIRECT != null) && settings.ENABLE_HTTP_REDIRECT;
+
+  udpPorts =
+    optional (settings.UDP_MUX_PORT != null) settings.UDP_MUX_PORT
+    ++ optional (settings.UDP_WHEP_PORT != null) settings.UDP_WHEP_PORT
+    ++ optional (settings.UDP_WHIP_PORT != null) settings.UDP_WHIP_PORT;
+
+  tcpPorts =
+    [ httpPort ]
```

Comment:
I am sure some people would like to run this behind a reverse proxy, so maybe this should be gated behind an additional option?